packages/devtools_app/lib/src/screens/deep_link_validation/validation_details_view.dart

```diff
@@ -495,63 +481,113 @@ class _PathCheckTable extends StatelessWidget {
-            collapsedBackgroundColor:
-                Theme.of(context).colorScheme.alternatingBackgroundColor2,
-            title: Row(
-              children: [
-                const SizedBox(width: defaultSpacing),
-                const Expanded(child: Text('Android')),
-                const Expanded(child: Text('IntentFiler')),
-                Expanded(
-                  child: intentFilterErrorCount > 0
-                      ? Text(
-                          '$intentFilterErrorCount Check failed',
-                          style: TextStyle(
-                            color: Theme.of(context).colorScheme.error,
-                          ),
-                        )
-                      : noIssueText,
-                ),
-              ],
-            ),
-            children: <Widget>[
-              for (final error in intentFilterErrors)
-                if (linkData.pathErrors.contains(error))
-                  Text(error.description),
-            ],
-          ),
+          _IntentFilterCheck(controller: controller),
           const Divider(height: 1.0),
-          ExpansionTile(
-            backgroundColor:
-                Theme.of(context).colorScheme.alternatingBackgroundColor2,
-            collapsedBackgroundColor:
-                Theme.of(context).colorScheme.alternatingBackgroundColor2,
-            title: Row(
-              children: [
-                const SizedBox(width: defaultSpacing),
-                const Expanded(child: Text('Android')),
-                const Expanded(child: Text('URL format')),
-                Expanded(
-                  child: linkData.pathErrors.contains(PathError.pathFormat)
-                      ? Text(
-                          'Check failed',
-                          style: TextStyle(
-                            color: Theme.of(context).colorScheme.error,
-                          ),
-                        )
-                      : noIssueText,
-                ),
-              ],
-            ),
-            children: <Widget>[
-              Text(PathError.pathFormat.description),
-            ],
-          ),
+          _PathFormatCheck(controller: controller),
         ],
       ),
     );
   }
 }
+
+class _IntentFilterCheck extends StatelessWidget {
+  const _IntentFilterCheck({required this.controller});
+
+  final DeepLinksController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    final linkData = controller.selectedLink.value!;
+    final theme = Theme.of(context);
+    final intentFilterErrorCount = intentFilterErrors
+        .where((error) => linkData.pathErrors.contains(error))
+        .toList()
+        .length;
+
+    return _PathCheckExpansionTile(
+      checkName: 'IntentFiler',
+      status: intentFilterErrorCount > 0
+          ? Text(
+              '$intentFilterErrorCount Check failed',
+              style: TextStyle(
+                color: theme.colorScheme.error,
+              ),
```

Comment:
can you add an extension method (getter) here https://github.com/flutter/devtools/blob/master/packages/devtools_app_shared/lib/src/ui/theme/theme.dart#L350
```
TextStyle get errorTextStyle => regularTextStyleWithColor(colorScheme.error);
```

Then this can be used here, below, and elsewhere where we are duplicating this text style. This will ensure that the styles from `regularTextStyle` are still applied to error text.